backend/src/main/java/net/vidux/camhub/discovery/CameraScanViduxHelper.java

```diff
@@ -0,0 +1,28 @@
+package net.vidux.camhub.discovery;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeoutException;
+
+class CameraScanViduxHelper implements CameraScan {
+
+  @Override
+  public CompletableFuture<Set<RawCameraData>> scanCams() {
+    List<String> rawCameraLineList;
+    try {
+      rawCameraLineList = new ViduxHelperRunner().callViduxHelper();
+    } catch (IOException | TimeoutException e) {
+      return CompletableFuture.failedFuture(e);
+    }
+
+    Set<RawCameraData> rawCameraSet = new HashSet<>();
+    RawCameraDataFactory factory = new RawCameraDataFactory();
+    for (String rawCameraLine : rawCameraLineList) {
+      rawCameraSet.add(factory.createRawCameraData(rawCameraLine));
+    }
```

Comment:
Ezt egyébként szerintem valahogy így is meg lehet csinálni és pár sort spórolhatunk:
`rawCameraLineList.stream().map(factory::createRawCameraData).collect(Collectors.toSet());`

vagy ha mindenképpen HashSet-et akarunk használni, akkor így:
`rawCameraLineList.stream().map(factory::createRawCameraData).collect(Collectors.toCollection(HashSet::new));`

A factory elnevezést amúgy nem javaslom. Ahogy Gábor is írta, legyen injektálva a RawCameraDataFactory és az legyen egyértelműen elnevezve (pl.:rawCameraDataFactory).